mitmproxy/tools/web/app.py

```diff
@@ -499,7 +499,7 @@ def __init__(self, master: "mitmproxy.tools.web.master.WebMaster", debug: bool)
         self.add_handlers("dns-rebind-protection", [(r"/.*", DnsRebind)])
         self.add_handlers(
             # make mitmweb accessible by IP only to prevent DNS rebinding.
-            r'^(localhost|[0-9.:\[\]]+)$',
+            r'^(localhost|[\da-f.:\[\]]+)$',
```

Comment:
I think this would allow `abcdef.ac` which is a valid domain + TLD - and therefore not really what we want here?

Reply:
It's incomplete (it validated "999...1234.123" as an IP for example) as it was before.
A full RE (OP provided one in #4072), which reliably validates all IPv4 and 6, will be really complicated 
(imho, it shouldn't rely on a regex for this purpose).

Reply:
I also think that this is quite a bullshit and you are transforming a "normal" issue to a security issue...

Please use the longer regex pattern - i am pretty sure that it wont cause issues!

Reply:
I get this is the first time you submit an issue. You should know this language is way outside the line.
Feel free to review and put the big RE. 
I'm out.